tests/phpunit/tests/media.php

```diff
@@ -3863,6 +3863,48 @@ public function test_wp_filter_content_tags_does_not_lazy_load_images_in_header(
 		$this->assertSame( '<div class="wp-site-blocks">' . $expected_template_content . '</div>', $html );
 	}
 
+	/**
+	 * @ticket 58089
+	 *
+	 * @covers ::wp_filter_content_tags
+	 * @covers ::wp_get_loading_attr_default
+	 */
+	public function test_wp_filter_content_tags_does_not_lazy_load_special_images_within_the_content() {
+		global $wp_query, $wp_the_query;
+
+		// Force no lazy-loading on the image tag expected in the content.
+		$expected_content = wpautop( wp_get_attachment_image( self::$large_id, 'large', false, array( 'loading' => false ) ) );
+
+		// Overwrite post content with an image.
+		add_filter(
+			'the_content',
+			function() {
+				// Replace content with an image tag, i.e. the 'wp_get_attachment_image' context is used while running 'the_content' filter.
+				return wp_get_attachment_image( self::$large_id, 'large', false );
+			},
```

Comment:
I think that's usually older unit tests from times where we still supported PHP 5.2 which didn't allow for closures. There's not much benefit to using named methods for test helpers, IMO it makes the tests harder to read because it moves some tightly coupled logic into separate methods. So I'd prefer to use closures for little things like this, unless we want to reuse it between multiple tests.